text/0000-compose.md

```diff
@@ -0,0 +1,100 @@
+* 2019-03-18
+* RFC PR: (leave this empty)
+* Fusion Issue: (leave this empty)
+
+# Summary
+
+This adds support for composing plugins via `app.compose`, and updates the `createToken` API to
+add an optional compose function as the second argument.
+
+# Basic example
+
+```js
+const ComposeExampleToken = createToken('ComposeExampleToken', (items) => {
+  return items.reduce((sum, item) => {
+    return sum + item;
+  }, 0);
+});
+
+app.compose(ComposeExampleToken, 2);
+app.compose(ComposeExampleToken, 3);
+
+app.register(createPlugin({
+  deps: {
+    composeExample: ComposeExampleToken
+  },
+  provides: ({composeExample}) {
+    console.log(composeExample); // 5
+  }
+}));
+```
+
+# Motivation
+
+There are often situations where you need to manage composing a large amount of objects onto a single token. This can be done today with a few approaches today,
+but they all require significant boilerplate and often unreasonable overheard. For example, you may want to split your graphql schemas and resolvers up into many
+separate modular schemas, and combine them onto a single token using schema stiching. However as the number of separate schemas expands upwards of 10, 20, it 
+becomes restrictive to manage separate tokens for each separate schema. For example:
+
+```js
+app.register(TokenA, SchemaA);
+app.register(TokenB, SchemaB);
+app.register(TokenC, SchemaC);
+app.register(TokenD, SchemaD);
+app.register(TokenE, SchemaE);
+
+app.register(SchemaToken, createPlugin({
+  deps: {
+    a: TokenA,
+    b: TokenB,
+    c: TokenC,
+    d: TokenD,
+    e: TokenE,
+  },
+  provides({a, b, c, d, e}) {
+    return mergeSchemas([a,b,c,d,e])
+  }
+}))
+```
+
+This becomes exceedingly difficult as the number of schemas increases. With the app.compose method, this could be replaced with:
```

Comment:
As noted above, this example also shifts the composition logic from the "composition plugin" to the token.